ddl/schema.go

```diff
@@ -54,13 +49,17 @@ func onCreateSchema(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error
 		if err != nil {
 			return ver, errors.Trace(err)
 		}
+		ver, err = updateSchemaVersion(t, job)
+		if err != nil {
+			return ver, errors.Trace(err)
+		}
```

Comment:
updateSchemaVersion and the check logic is in the same transaction t, they could be done atomically. 
That's means the updateScheamVersion + DropSchema + GetIDs() are atomic, either all succeed or all fail.

Reply:
I did not mean it is not atomic. But I want to call `result = getIDs()` first before I set `job.CtxVars = []interface{}{result};updateSchemaVersion(...);`.

Reply:
Why do you need the result of `getIDs`? You can also get it by `ListTables` after execution, in `ApplyDiff`.

Reply:
I need to invoke `infosync.PutRuleBundles` to drop rules anyway. I don't want to get it twice: once in the DDL jobs, and another in `ApplyDiff`.

And, of course, I can run `ListTables`. It is indeed better that it did not use the confusing `AffectedOpts`.

Then, the only point for this PR is consistent behavior/usage of `updateSchemaVersion`. If you think that is unnecessary, I will close the PR.